### PR TITLE
Replace unnecesary sprintf with sprint

### PR DESCRIPTION
**Pull request by fiunchinho**
AFAIK `Sprintf` is not necessary when already using `Debugf`.

### CODE REVIEW

**Review by axbarsan — APPROVED** [review]

File: service/controller/cluster/handler/clusterownerreference/controlplane.go [hunk 1]
```diff
@@ -46,7 +45,7 @@ func (r *Resource) ensureControlPlane(ctx context.Context, cluster *capi.Cluster
 }
 
 func (r *Resource) updateControlPlaneObject(ctx context.Context, cluster *capi.Cluster, azureMachine *capz.AzureMachine) error {
-	r.logger.Debugf(ctx, "message", fmt.Sprintf("ensuring %s label and 'ownerReference' fields on AzureMachine '%s/%s'", capi.ClusterLabelName, azureMachine.Namespace, azureMachine.Name))
+	r.logger.Debugf(ctx, "message", "ensuring %s label and 'ownerReference' fields on AzureMachine '%s/%s'", capi.ClusterLabelName, azureMachine.Namespace, azureMachine.Name)
```
Comment (tuommaki) [inline service/controller/cluster/handler/clusterownerreference/controlplane.go:48]: The `"message", ` part should be snapped from these. The first string after `context.Context` is the `message` :slightly_smiling_face:

**Comment by fiunchinho** [comment]
Ok, PTAL now

File: service/controller/cluster/handler/clusterownerreference/controlplane.go [hunk 2]
```diff
@@ -61,14 +60,14 @@ func (r *Resource) updateControlPlaneObject(ctx context.Context, cluster *capi.C
 
 	err = r.ctrlClient.Update(ctx, azureMachine)
 	if apierrors.IsConflict(err) {
-		r.logger.Debugf(ctx, "message", "conflict trying to save object in k8s API concurrently", "stack", microerror.JSON(microerror.Mask(err)))
-		r.logger.Debugf(ctx, "message", "canceling resource")
+		r.logger.Debugf(ctx, "conflict trying to save object in k8s API concurrently", "stack", microerror.JSON(microerror.Mask(err)))
```
[the remark this answers is not in the document]
Comment (fiunchinho) [reply service/controller/cluster/handler/clusterownerreference/controlplane.go:63]: mm.... What should we do here then? If I use `Errorf` we will log an error in a kind-of-expected scenario. I believe the best would be to keep the warning, and ditch the stack: we already know it was a conflict error.

[the remark this answers is not in the document]
Comment (fiunchinho) [reply service/controller/cluster/handler/clusterownerreference/controlplane.go]: I pushed commit with that approach.

**Review by tuommaki — APPROVED** [review]

File: service/controller/cluster/handler/clusterownerreference/controlplane.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tuommaki) [reply service/controller/cluster/handler/clusterownerreference/controlplane.go]: Yeah, that works.

**Pull request merged**